### PR TITLE
Support root rotation

### PR DESCRIPTION
**Pull request by hosseinsia**
Update the root according to 
https://theupdateframework.github.io/specification/v1.0.19/index.html#load-trusted-root

### CODE REVIEW

**Comment by hosseinsia** [comment]
Fixtures are generated using this repo:
https://github.com/DataDog/tuf-fixtures

File: client/client.go [hunk 1]
```diff
@@ -249,6 +264,138 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
...
+	if err := c.getLocalMeta(); err != nil {
+		if _, ok := err.(verify.ErrExpired); !ok {
+			return err
+		}
+	}
+
+	// Prepare for 5.3.11: If the timestamp and / or snapshot keys have been rotated,
+	// then delete the trusted timestamp and snapshot metadata files.
+	getKeyIDs := func(role string) []string {
+		keyIDs := make([]string, 0, len(c.db.GetRole("timestamp").KeyIDs))
```
Comment (trishankatdatadog) [inline client/client.go:284]:
```suggestion
		keyIDs := make([]string, 0, len(c.db.GetRole(role).KeyIDs))
```

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]
Will send more comments later, but this is really getting into shape, thanks!

File: client/client.go [hunk 2]
```diff
@@ -249,6 +264,138 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
...
+	// matter, because we will attempt to update it in the next step.
+	if err := c.getLocalMeta(); err != nil {
+		if _, ok := err.(verify.ErrExpired); !ok {
+			return err
+		}
+	}
+
+	// Prepare for 5.3.11: If the timestamp and / or snapshot keys have been rotated,
+	// then delete the trusted timestamp and snapshot metadata files.
+	getKeyIDs := func(role string) []string {
```
Comment (trishankatdatadog) [inline client/client.go:283]:
```suggestion
	getKeyIDs := func(rolename string) []string {
```

File: verify/errors.go [hunk 3]
```diff
@@ -50,6 +50,15 @@ func (e ErrLowVersion) Error() string {
 	return fmt.Sprintf("version %d is lower than current version %d", e.Actual, e.Current)
 }
 
+type ErrWrongVersion struct {
+	Given    int
+	Expected int
+}
+
+func (e ErrWrongVersion) Error() string {
+	return fmt.Sprintf("version %d does not match the extepcted version %d", e.Given, e.Expected)
```
Comment (lebauce) [inline verify/errors.go:59]: s/extepcted/expected/

File: verify/verify.go [hunk 4]
```diff
@@ -118,6 +118,29 @@ func (db *DB) Unmarshal(b []byte, v interface{}, role string, minVersion int) er
 	return json.Unmarshal(s.Signed, v)
 }
 
+// UnmarshalExpired is exactly like Unmarshal except ignores expired timestamp.
+func (db *DB) UnmarshalExpired(b []byte, v interface{}, role string, minVersion int) error {
+	s := &data.Signed{}
+	if err := json.Unmarshal(b, s); err != nil {
+		return err
+	}
+	// Note: If verification fails, then we wont attemp to unmarshal
```
Comment (lebauce) [inline verify/verify.go:127]: s/attemp/attempt/

File: verify/verify.go [hunk 5]
```diff
@@ -118,6 +118,29 @@ func (db *DB) Unmarshal(b []byte, v interface{}, role string, minVersion int) er
...
+	verifyErr := db.Verify(s, role, minVersion)
+	if verifyErr != nil {
+		if _, ok := verifyErr.(ErrExpired); !ok {
+			return verifyErr
+		}
+	}
+	if err := json.Unmarshal(s.Signed, v); err != nil {
+		return err
+	}
+	if verifyErr != nil {
```
Comment (lebauce) [inline verify/verify.go:138]: return verifyErr

**Comment by hosseinsia** [comment]
This is an abandoned PR. We decided to update the go-tuf upstream now. Please review this instead:
https://github.com/theupdateframework/go-tuf/pull/143

**Pull request closed without merging**